liblangutil/Token.cpp

```diff
@@ -72,6 +72,9 @@ void ElementaryTypeNameToken::assertDetails(Token _baseType, unsigned const& _fi
 			"No elementary type " + string(TokenTraits::toString(_baseType)) + to_string(_first) + "x" + to_string(_second) + "."
 		);
 	}
+	else if (_baseType == Token::Address)
+		solAssert(_first == 0 && _second == 0, "Address should not have size arguments");
```

Comment:
Do you want to pull this out into another PR so at least we can merge this? I think it belongs in its own anyway.

Reply:
Sure, I can do that. (We can merge this PR anyway though)

Reply:
#7923